hacks/generateOptimizeSidebars.js

```diff
@@ -74,22 +89,70 @@ function buildLi(li) {
   }
 }
 
-// The root of the sidebar menu.
-const root = document.querySelector(".theme-doc-sidebar-menu");
-
 // Generates the sidebar definitions for the entire menu.
 function buildSidebars() {
+  // The root of the sidebar menu.
+  const root = document.querySelector(".theme-doc-sidebar-menu");
+  if (root === null) {
+    throw new Error("Unable to find sidebar menu root.");
+  }
+
   const items = Array.from(root.querySelectorAll(":scope > li"));
 
   const mappedItems = items.map(buildLi);
 
-  return `{
-    "Components": [
+  return `[
       ${mappedItems.join("\n")}
-    ]
+    ]`;
+}
+
+// Identifies configuration strategy based on current URL.
+function identifyStrategyFromPath() {
+  const path = window.location.pathname;
+  const strategyKey = Object.keys(strategies).find((strategyPath) =>
+    path.startsWith(strategyPath)
+  );
+
+  if (strategyKey === undefined) {
+    throw new Error(
+      "I'm not sure how to generate sidebars for this URL. Are you sure you're on the correct page?"
+    );
   }
-  `;
+  strategy = strategies[strategyKey];
 }
 
-// Copies the generated sidebar definitions to the clipboard.
-copy(buildSidebars());
+// Expands all sidebar items (so that we can walk them all in the DOM).
+async function expandAllSidebarItems() {
+  const count = () =>
+    document.querySelectorAll(".menu__link--sublist[aria-expanded=false]")
+      .length;
+  const expand = () =>
+    document
+      .querySelectorAll(".menu__link--sublist[aria-expanded=false]")
+      /** @ts-ignore */
+      .forEach((x) => x.click());
```

Comment:
ts-ignore'd because there is a type-casting I need to do to `x` to make the `click` method available...but I couldn't figure out what it was. It didn't seem important for me to figure it out.

Reply:
Do you want to leave yourself a note or create an issue for this?

Reply:
Weird, I thought I responded to this one but I must never have hit the button.

I'm less concerned about a comment here -- if someone is ambitious, they'll remove the ts-ignore, and see the error message that describes why I added it in the first place.